src/service.rs

```diff
@@ -397,16 +397,30 @@ impl Service {
     }
 }
 
+impl Drop for Service {
+    fn drop(&mut self) {
+        // Disconnect from all peers when we drop the service
+        let mut cm = unwrap_result!(self.connection_map.lock());
+        cm.clear();
+    }
+}
+
```

Comment:
This change was already introduced in this PR: https://github.com/maidsafe/crust/pull/537/files#diff-bfa17b8ec2b0aac77da5f96cbcbe4101R400